content/en/blog/2022-11-11-prove-the-authenticity-of-helm-charts/index.md

```diff
@@ -0,0 +1,306 @@
+
+For the _HelmRepository_ resource:
+
+```shell
+$ flux get sources helm
+NAME                    REVISION        SUSPENDED       READY   MESSAGE
+prometheus-community                    False           True    Helm repository is ready
+```
+
+For the _HelmRelease_ resource:
+
+```shell
+$ flux get helmreleases
+NAME            REVISION        SUSPENDED       READY   MESSAGE
+prometheus      15.18.0         False           True    Release reconciliation succeeded
+```
+
+If everything is fine, you can check the pods in the `monitoring` namespace:
+
+```shell
+$ kubectl get pods -n monitoring
+NAME                                             READY   STATUS    RESTARTS   AGE
+prometheus-alertmanager-54b7d7cf45-2b7zf         2/2     Running   0          115s
+prometheus-kube-state-metrics-67f68d64bb-vlmvd   1/1     Running   0          115s
+prometheus-node-exporter-46gm6                   1/1     Running   0          115s
+prometheus-pushgateway-596cd99697-t79zt          1/1     Running   0          115s
+prometheus-server-c458cf6f9-nvstw                2/2     Running   0          115s
+```
+
+Great! Now, you have installed Prometheus with Flux by using the Helm chart stored in the OCI registry and verified it with _cosign_.
+
+We can assume that the Helm chart's signature is verified as we let it be deployed in a cluster but let's do have double check and check the status of the _HelmRelease_ to see whether the verification is successful or not:
+
+```shell
+$ kubectl get helmcharts -n flux-system flux-system-prometheus -ojsonpath='{.status.conditions[?(@.type=="SourceVerified")]}'
+{"lastTransitionTime":"2022-11-09T13:27:38Z","message":"verified signature of version 15.18.0","observedGeneration":1,"reason":"Succeeded","status":"True","type":"SourceVerified"}
+```
+
+That's super cool! Because Flux is going to add a condition to the HelmChart resource's status section to show the verification status. If the verification is successful, it will add a condition like the one above.
+
+## DIY (Do it yourself) Approach
+
+The Prometheus community Helm charts only serve as an example. Here is how you can do the same thing with your own Helm charts.
+
+1. Create a Helm chart
+2. Package the Helm chart as .tar.gz file
+3. Login to the OCI-compliant registry that you want to use to store your Helm chart
+4. Push the Helm chart as OCI artifact
+
+Let's create a sample directory that will contain our Helm chart:
+
+```shell
+mkdir -p helm-oci-demo
+cd helm-oci demo
+```
+
+Now package the Helm chart as .tar.gz file:
+
+```shell
+ helm create nginx
+```
+
+Let's package the Helm chart as .tar.gz file:
+
+```shell
+helm package nginx
+```
+
+Now, let's login to the OCI-compliant registry that you want to use to store your Helm chart. In this example, we'll be using the GitHub Container Registry (ghcr.io):
+
+```shell
+helm registry login ghcr.io -u $USER -p $PASSWORD
```

Comment:
@developer-guy Would it be better to suggest using a PAT with packages write scope, like this? We want to encourage 2FA, and you can't log in with username and pass on the CLI if you have that set.

```suggestion
echo $GHCR_PAT | docker login ghcr.io -u $USER --password-stdin
```

Reply:
That's an amazing suggestion, thanks for bringing it my attention, fixed.